AsyncDisplayKit/Details/Transactions/_ASAsyncTransactionContainer.m

```diff
@@ -102,19 +88,18 @@ - (_ASAsyncTransaction *)asyncdisplaykit_asyncTransaction
       transactions = [NSHashTable hashTableWithOptions:NSPointerFunctionsObjectPointerPersonality];
       self.asyncdisplaykit_asyncLayerTransactions = transactions;
     }
+    __weak CALayer *weakSelf = self;
     transaction = [[_ASAsyncTransaction alloc] initWithCallbackQueue:dispatch_get_main_queue() completionBlock:^(_ASAsyncTransaction *completedTransaction, BOOL cancelled) {
+      __strong CALayer *self = weakSelf;
+      if (self == nil) {
+        return;
+      }
```

Comment:
Thoughts on this __weak here @appleguy ? Zeroing weak refs aren't free but I don't like the idea of all of our layers establishing retain cycles during their display. And given the issue we saw today, it definitely causes layers to be retained longer than their nodes in some cases.